src/transformStateWithClones.js

```diff
@@ -7,7 +7,45 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(stateCopy, action.extraData);
+        stateHistory.push({ ...stateCopy });
+        break;
+
+      case 'removeProperties':
+        removeProperties(stateCopy, action.keysToRemove);
+        stateHistory.push({ ...stateCopy });
+        break;
+
+      case 'clear':
+        clearProperties(stateCopy);
+        stateHistory.push({ ...stateCopy });
```

Comment:
[DONT REPEAT YOURSELF] - You are pushing to 'stateHistory' in every case of the switch statement. This action should be performed once after the switch block, to avoid repetition.

Reply:
<img width="797" alt="image" src="https://github.com/user-attachments/assets/5168b4fb-fa9f-4e01-8731-6486c39cc63c">